### PR TITLE
chore: make dashboards filter term persist across pages

### LINKED ISSUE #844: Remember the filter for the list pages
the experience today is that you go to the tasks page (or dashboards page), use the filter to find the tasks you want, then click on one of them. When you click the back button in the browser, you are back to the page with no filters in place.

Users can have 100s of tasks, so i'd love to be able to save that filter and get back to it.

I propose we find a way to keep the filter in place when a user clicks the back button specifically for the tasks page.

### PR DESCRIPTION
**Pull request by subirjolly**
Closes #844 

**Before**
before

**After**
after

### CODE REVIEW

File: src/dashboards/components/dashboard_index/DashboardsIndex.tsx [hunk 1]
```diff
@@ -104,7 +95,7 @@ class DashboardIndex extends PureComponent<Props, State> {
               <GetAssetLimits>
                 <DashboardsIndexContents
                   searchTerm={searchTerm}
-                  onFilterChange={this.handleFilterDashboards}
+                  onFilterChange={this.props.setSearchTerm}
```
Comment (Asalem1) [inline src/dashboards/components/dashboard_index/DashboardsIndex.tsx:98]: if the `onSearch` and `onFilterChange` are being mapped to props, there's no reason for us to prop drill. We could just map it out directly to the child

File: src/dashboards/components/dashboard_index/DashboardsIndex.tsx [hunk 2]
```diff
@@ -169,16 +156,19 @@ const mstp = (state: AppState) => {
     cloud: {limits},
   } = state
   const sortOptions = state.resources.dashboards['sortOptions']
+  const searchTerm = state.resources.dashboards['searchTerm']
```
Comment (Asalem1) [inline src/dashboards/components/dashboard_index/DashboardsIndex.tsx:159]: nit: We can probably consolidate these to a `getAll`, then just get the individual properties:

https://github.com/influxdata/ui/blob/master/src/resources/selectors/index.ts#L14

So something like:

```
const {sortOptions, searchTerm} = getAll<Dashboard>(
    state,
    ResourceType.Dashoards
  )
```

**Review by Asalem1 — CHANGES_REQUESTED** [review]
Besides the comments, my concern with this approach is that we are now dispatching an event whenever we search through dashboards. This might cause us to rerender the DOM a bunch, which might cause a performance hit. Is there any chance we can get a performance chart to see the difference between this implementation and the original? I know the original isn't what we're striving for, but it might be possible for us to use that as a barometer for success here when measuring how performant this part of the app currently is vs how it fairs with the refactor

File: cypress/e2e/shared/dashboardsIndex.test.ts [hunk 3]
```diff
@@ -481,6 +481,36 @@ describe('Dashboards', () => {
...
+          .first()
+          .getByTestID('dashboard-card--name')
+          .should('contain', dashboardName)
+
+        // Navigate Away and come back using browser's back button
+        cy.getByTestID('dashboard-card')
+          .first()
+          .getByTestID('dashboard-card--name')
+          .click()
+        cy.wait(1000)
```
Comment (joelfried) [inline cypress/e2e/shared/dashboardsIndex.test.ts:502]: Can we handle the navigation without these hard coded wait here and below?  (And looking [here](https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting) I'm not sure if both of these are necessary.  Perhaps you need to add a cy.get after the click on 501 before you go back though? Not sure on that.)

**Review by joelfried — APPROVED** [review]

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 60 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 91 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1160)

File: src/dashboards/reducers/dashboards.ts [hunk 4]
```diff
@@ -150,6 +152,14 @@ export const dashboardsReducer = (
 
         return
       }
+
+      case SET_DASHBOARDS_SEARCH_TERM: {
+        const {searchTerm} = action
+
+        draftState['searchTerm'] = searchTerm
```
Comment (Asalem1) [inline src/dashboards/reducers/dashboards.ts:159]: nit: for stylistic consistencies let's use dot notation

**Review by Asalem1 — APPROVED** [review]
Looks great. Happy to get this in first thing Monday  morning

**Pull request merged**